formatTest/typeCheckedTests/expected_output/comments.re

```diff
@@ -21,43 +34,44 @@
 /* comment ***/
 /* comment ****/
 /* comment *****/
+
 /**
  * Multiline
  */
+
 /** Multiline
  *
  */
+
 /**
  **
  */
-
 let testingNotQuiteEndOfLineComments = [
-  "Item 1", /* Comment For First Item */
-  "Item 2", /* Comment For Second Item */
-  "Item 3", /* Comment For Third Item */
+  "Item 1" /* Comment For First Item */,
+  "Item 2" /* Comment For Second Item */,
+  "Item 3" /* Comment For Third Item */,
   "Item 4" /* Comment For Fourth Item - but no semi */
   /* Comment after last item in list. */
-];
+] /* Comment after list bracket */;
 
-/* Comment after list bracket */
 let testingEndOfLineComments = [
   "Item 1", /* Comment For First Item */
   "Item 2", /* Comment For Second Item */
   "Item 3", /* Comment For Third Item */
   "Item 4" /* Comment For Fourth Item - but before semi */
   /* Comment after last item in list. */
-];
+] /* Comment after list bracket */;
 
-/* Comment after list bracket */
 /* This time no space between bracket and comment */
-let testingEndOfLineComments = [];
+let testingEndOfLineComments =
+  [] /* Comment after list bracket */;
 
-/* Comment after list bracket */
-type t = (int, int); /* End of line on t */
+type t = (int, int /* End of line on t */);
```

Comment:
You should see what happens when in Reason you format:

```
type t = (int, int);  /* End of line on t */
```

In that case, is it preserved? There are more places for a comment to exist in Reason, and as long as each of them is preserved in Reason, it's pretty acceptable for converting from OCaml to behave this way.

However, the old code gave priority to the comment being on the entire type definition, which is why it was correctly formatted the way you might expect. The heuristic to preserving this is pretty simple. If the "attachment location" of the comment is an entire top level definition (includes the entire definition) and the comment is on the end of a line, then we render it as an end of line comment on the entire definition.